### PR TITLE
SDKS-80 SSL Pinning Implementation

### PR DESCRIPTION
**Pull request by witrisna**

### CODE REVIEW

File: forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java [hunk 1]
```diff
@@ -230,7 +233,102 @@ public void onResponse(@NotNull Call call, @NotNull Response response) throws IO
...
+
+        Request request = new Request.Builder()
+                .get()
+                .url("https://api.ipify.org?format=json")
+                .build();
+
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        final boolean[] result = new boolean[1];
+
```
Comment (jeyanthanperiyasamy) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java:270]: @witrisna  can we have a result[0] = false;

**Review by jeyanthanperiyasamy — COMMENTED** [review]
proposing a small change
https://github.com/ForgeRock/forgerock-android-sdk/pull/145/files#r816167174

File: forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java:270]: The default value for boolean primitive type is false, any specific reason we need to set the value?

File: forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
```
Comment (rodrigoareis) [inline forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java:2]: Copyright (c) 2022 ForgeRock

File: forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                .pin("sha1/2vB3hhEJ98C5efhhWpxtD2wxYek=")
+                .build();
+        OkHttpClient client = OkHttpClientProvider.getInstance().lookup(networkConfig);
+        //Make sure the pinning is configured
+        Assertions.assertThat(client.certificatePinner().getPins())
+                .contains(new CertificatePinner.Pin(server.getHostName(),
+                        "sha256/9hNxmEFgLKGJXqgp61hyb8yIyiT9u0vgDZh4y8TmY/M="))
+                .contains(new CertificatePinner.Pin(server.getHostName(), "sha1/2vB3hhEJ98C5efhhWpxtD2wxYek="));
+    }
+
```
Comment (rodrigoareis) [inline forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java:91]: Do you think is valid to include unit tests public keys from different types of Certificates?

File: samples/auth/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -13,47 +13,67 @@
...
     <integer name="forgerock_oauth_threshold" translatable="false">30</integer> <!-- in second -->
 
     <!-- The following setting is for local development environment only -->
     <!-- Server -->
-    <string name="forgerock_url" translatable="false">http://openam.example.com/openam</string>
+    <!--
+    <string name="forgerock_url" translatable="false">http://192.168.86.23:8081/openam</string>
+    <string name="forgerock_url" translatable="false"> https://openam-forgerrock-sdksregression.forgeblocks.com/am</string>
+    -->
+    <string name="forgerock_url" translatable="false">https://192.168.86.23:9443/openam</string>
```
Comment (rodrigoareis) [inline samples/auth/src/main/res/values/strings.xml:38]: Should we change it to a fake host?

File: samples/auth/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -13,47 +13,67 @@
...
 
     <!-- Service -->
-    <string name="forgerock_auth_service" translatable="false">Test</string>
+    <string name="forgerock_auth_service" translatable="false">webauthn</string>
     <string name="forgerock_registration_service" translatable="false">Registration Tree2</string>
 
-    <!-- WebAuthn Biometric -->
     <string name="asset_statements" translatable="false">           [{
-             "include": "https://your-host/.well-known/assetlinks.json"
+             "include": "https://default.forgeops.petrov.ca/.well-known/assetlinks.json"
```
Comment (rodrigoareis) [inline samples/auth/src/main/res/values/strings.xml:67]: Same as above

File: samples/auth/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -13,47 +13,67 @@
...
            }]</string>
 
     <!-- Facebook -->
-    <string name="facebook_app_id">your facebook app id</string>
-    <string name="fb_login_protocol_scheme">your facebook scheme</string>
+    <string name="facebook_app_id">622429077955089</string>
```
Comment (rodrigoareis) [inline samples/auth/src/main/res/values/strings.xml:71]: Revert to instruction

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Overall implementation looks good me. Left some comments to be addressed

File: forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java (hunk 3 above)
Comment (witrisna) [reply forgerock-core/src/test/java/org/forgerock/android/auth/NetworkConfigTest.java:91]: This test is more focused on the configuration test.
The pinning test can be found under:
org.forgerock.android.auth.ServerConfigTest#testSha1Pinning
org.forgerock.android.auth.ServerConfigTest#testSha256Pinning

File: samples/auth/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -13,47 +13,54 @@
...
     <string name="asset_statements" translatable="false">           [{
-             "include": "https://your-host/.well-known/assetlinks.json"
+             "include": "https://openam.example.com/.well-known/assetlinks.json"
            }]</string>
 
     <!-- Facebook -->
     <string name="facebook_app_id">your facebook app id</string>
     <string name="fb_login_protocol_scheme">your facebook scheme</string>
 
+    <string name="apple_scheme">org.forgerock.demo</string>
```
Comment (spetrov) [inline samples/auth/src/main/res/values/strings.xml:62]: Is that a new property? I didn't find its usage anywhere in the project...

**Review by spetrov — CHANGES_REQUESTED** [review]

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by spetrov — APPROVED** [review]
👍🏻

**Pull request merged**